### PR TITLE
Implement warg component registry support.

### PR DESCRIPTION
**Pull request by peterhuene**
This PR implements support for warg component registries.

It also removes support for "local" registries and, subsequently, the `registry` subcommand.

The following commands have been implemented:

* `publish` - publishes the current component to a registry.
* `signing new-key` - create a new signing key in the system keyring.
* `signing set-key` - set a signing key in the system keyring.
* `signing delete-key` - delete a signing key from the system keyring.

The tests now require a `warg-server` be on the path to run a local, temporary component registry.

### CODE REVIEW

File: src/registry.rs [hunk 1]
```diff
@@ -434,228 +445,574 @@ impl DependencyResolution {
...
-            registry: Box<dyn Registry>,
-            packages: HashMap<&'a PackageId, Vec<PackageEntry<'a>>>,
-        }
+        Ok((self.target_resolutions, self.resolutions))
     }
 
-    /// Iterates over all dependencies of the package.
-    pub fn deps(&self) -> impl Iterator<Item = (&String, &DependencyResolution)> {
-        self.target_dependencies
+    async fn update_and_download(&mut self) -> Result<()> {
```
Comment (Kylebrown9) [inline src/registry.rs:867]: Separate out update and download into their own functions that pipe data through.

File: src/commands/signing.rs
Comment (Kylebrown9) [inline src/commands/signing.rs:1]: Would be more aptly named `key` than `signing`.

**Comment by Kylebrown9** [comment]
Thanks for the PR, it looks great!

File: src/commands/signing.rs
Comment (peterhuene) [reply src/commands/signing.rs:1]: The new commands are `key new`, `key set`, and `key delete`.

File: src/registry.rs (hunk 1 above)
Comment (peterhuene) [reply src/registry.rs]: I've split this up a little bit better where the update and "download and resolve" steps feed state through.

Let me know if the change doesn't make sense.

**Review by Kylebrown9 — APPROVED** [review]
Looks great, thanks!

**Pull request merged**